doc/oob_streams.md

```diff
@@ -47,14 +47,10 @@ The server may also reply with stream or pipe:
 ```cs
 public async Task<Stream> SendLargeFileAsync()
 {
-   var streamPair = FullDuplexPipe.CreatePair();
-   
-   // Use streamPair.Item1
+   // Open a stream on the server-side and forward it to the client.
+   var stream = new FileStream("pathToServerFile", FileMode.Open);
```

Comment:
Since the stream is used for async I/O, it's a significant perf improvement to open the stream stating that async I/O will be used, [according to the docs](https://docs.microsoft.com/en-us/dotnet/api/system.io.filestream.-ctor?view=netcore-3.1#System_IO_FileStream__ctor_System_String_System_IO_FileMode_System_IO_FileAccess_System_IO_FileShare_System_Int32_System_Boolean_):

> If the application is designed to take advantage of asynchronous I/O, set the useAsync parameter to true. Using asynchronous I/O correctly can speed up applications by as much as a factor of 10, but using it without redesigning the application for asynchronous I/O can decrease performance by as much as a factor of 10.

So I suggest we update the sample to:

```cs
public Stream GetFile(string path) {
    // Validate that the client should be granted access to the requested file.
    return new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, useAsync: true);
}
```

I realize this is just a sample in a doc, but as it will likely serve as a reference to folks actually writing such code, let's show them how to do it in a performant way.

Reply:
I very much agree, docs set a precedent for newcomers and the better it's written the better the average project that uses it will be. Fixed!